include/libembeddedhal/rotary_encoder/interface.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <cstdint>
+
+#include "../error.hpp"
+#include "../frequency.hpp"
+#include "../percent.hpp"
+
+namespace embed {
+/**
+ * @brief Hardware abstraction interface for Rotary Encoder feedback devices.
+ *
+ */
+class rotary_encoder
+{
+public:
+  /**
+   * @brief Return the current angular position from the feedback device.
+   *
+   * @return boost::leaf::result<nanoradian> - current angular position reading or an
+   * error.
+   */
+  [[nodiscard]] boost::leaf::result<angular_position> read() noexcept
+  {
+    return read();
```

Comment:
```suggestion
    return driver_read();
```
Otherwise this is recursive